src/main/java/uk/gov/hmcts/reform/civil/service/PaymentRequestUpdateCallbackService.java

```diff
@@ -3,167 +3,236 @@
+            log.info("FeeType {} is not handled. No action required.", feeType);
+            return;
+        }
+
+        Long caseId = Long.valueOf(dto.getCcdCaseNumber());
+        log.info("Fetching case details for caseId {}", caseId);
+        log.debug("ServiceRequestUpdateDto: {}", dto);
+
+        CaseDetails caseDetails = coreCaseDataService.getCase(caseId);
+        CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+
+        processPaymentUpdate(dto, caseData, feeType, caseId);
     }
 
-    private static boolean isValidPaymentUpdateHearing(String feeType, CaseData caseData) {
-        return feeType.equals(FeeType.HEARING.name())
-            && (caseData.getHearingFeePaymentDetails() == null || caseData.getHearingFeePaymentDetails().getStatus() == FAILED);
+    private void processPaymentUpdate(ServiceRequestUpdateDto dto, CaseData caseData, FeeType feeType, Long caseId) {
+        if (!caseData.isLipvLipOneVOne()) {
+            handlePaymentUpdate(dto, caseData, feeType, caseId, false);
+            return;
+        }
+
+        if (isPaymentUpdateValid(feeType, caseData)) {
+            handlePaymentUpdate(dto, caseData, feeType, caseId, true);
+        } else {
+            log.info("Payment update is not valid for case {} and feeType {}", caseId, feeType);
+        }
     }
 
-    private static boolean isValidUpdatePaymentClaimIssue(String feeType, CaseData caseData) {
-        return feeType.equals(FeeType.CLAIMISSUED.name())
-            && (caseData.getClaimIssuedPaymentDetails() == null || caseData.getClaimIssuedPaymentDetails().getStatus() == FAILED);
+    private boolean isPaid(ServiceRequestUpdateDto dto) {
+        return PAID.equalsIgnoreCase(dto.getServiceRequestStatus());
     }
 
-    private CardPaymentStatusResponse getCardPaymentStatusResponse(ServiceRequestUpdateDto serviceRequestUpdateDto) {
+    private FeeType getFeeType(String feeTypeStr) {
+        try {
+            return FeeType.valueOf(feeTypeStr);
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
 
-        return CardPaymentStatusResponse.builder().paymentReference(serviceRequestUpdateDto.getPayment().getPaymentReference())
-            .status(String.valueOf(SUCCESS)).build();
+    private boolean isHandledFeeType(FeeType feeType) {
+        return feeType == FeeType.HEARING || feeType == FeeType.CLAIMISSUED;
     }
 
-    private void createEvent(CaseData caseData, String caseId, String feeType) {
+    private boolean isPaymentUpdateValid(FeeType feeType, CaseData caseData) {
+        return (feeType == FeeType.HEARING && isHearingPaymentFailed(caseData))
+            || (feeType == FeeType.CLAIMISSUED && isClaimIssuePaymentFailed(caseData));
+    }
 
-        StartEventResponse startEventResponse = coreCaseDataService.startUpdate(
-            caseId,
-            getEventNameFromFeeType(caseData, feeType)
-        );
+    private boolean isHearingPaymentFailed(CaseData caseData) {
```

Comment:
This isn't quite accurate - it's either failed or null which means they won't have paid yet, so it's more like isUnpaidOrFailed() - don't want to confuse ppl - same with the other function like this.